### PR TITLE
Change Labels On Ontology Concepts

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

**Comment by fppt** [comment]
!rtg

File: grakn-core/src/main/java/ai/grakn/concept/OntologyConcept.java [hunk 1]
```diff
@@ -38,6 +38,14 @@
  * @author fppt
  */
 public interface OntologyConcept extends Concept {
+    //------------------------------------- Modifiers ----------------------------------
+    /**
+     * Changes the {@link Label} of this {@link Concept} to a new one.
+     * @param label The new {@link Label}.
+     * @return The {@link Concept} itself
+     */
+    OntologyConcept setLabel(Label label);
```
Comment (aelred) [inline grakn-core/src/main/java/ai/grakn/concept/OntologyConcept.java]: We need a description and `@throw` clause of what happens if the label in question is already in use, plus a test for it.

Comment (fppt) [inline grakn-core/src/main/java/ai/grakn/concept/OntologyConcept.java]: I have added the test. But I thought we agreed that trying to document runtime exceptions was just not maintainable in the long run?

Comment (aelred) [inline grakn-core/src/main/java/ai/grakn/concept/OntologyConcept.java]: In that case, a test is fine!

**Pull request merged**